java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/refactoring/MoveRefactoring.java

```diff
@@ -420,18 +420,26 @@ void doRefactoring(MoveElementUI ui) {
         }
 
         private static Project getSelectedProject(NamedPath selectedProject) {
+            String path = selectedProject == null ? null : selectedProject.getPath();
+            if (path == null) {
+                return null;
+            }
+            FileObject file;
             try {
-                String path = selectedProject.getPath();
-                return path != null ? FileOwnerQuery.getOwner(Utils.fromUri(path)) : null;
+                file = Utils.fromUri(path);
```

Comment:
I would prefer if the patch would be smaller (e.g. `String path = selectedProject != null ? selectedProject.getPath() : null` is a one file change, and another similar change below), but I can live with this.

Reply:
Hmm I feel the current one is slightly more clear so if it seems okay to you we can perhaps go with it.

Reply:
I am not sure why you rewrote the whole method, the worry is that `selectedProject` might be null, the rest is already checking for null.

so the original version can be kept with one addition:

```java
        private static Project getSelectedProject(NamedPath selectedProject) {
            if (selectedProject == null) {
                return null;
            }
            try {
                String path = selectedProject.getPath();
                return path != null ? FileOwnerQuery.getOwner(Utils.fromUri(path)) : null;
            } catch (MalformedURLException ex) {
                return null;
            }
        }
```

Reply:
Thanks @mbien ! We can go with one line change as you suggested . I will update the PR.